packages/x-data-grid-generator/src/hooks/useMockServer.ts

```diff
@@ -367,5 +368,6 @@ export const useMockServer = (
     loadNewData: () => {
       setIndex((oldIndex) => oldIndex + 1);
     },
+    isReady: Boolean(data?.rows?.length),
```

Comment:
It could probably be used in the demos too, let's leave it for later though.
Side note: When doing `loadNewData` it might not convey the correct information initially since the previous data would be there but outdated, isn't much of a problem though since updating the `fetchRows` would reset the `dataSource` retriggering a fetch call anyways.